File: solutions/uri/1049/1049.cpp
#include <cstdio>
#include <iostream>
#include <string>

int
    main() {
    std::string a, b, c;

    std::cin >> a;
    std::cin >> b;
    std::cin >> c;

    if (a == "vertebrado") {
        if (b == "ave") {
            if (c == "carnivoro")
                printf("aguia\n");
            else
                printf("pomba\n");
        } else {
            if (c == "onivoro") {
                printf("homem\n");
            } else {
                printf("vaca\n");
            }
        }
    } else {
        if (b == "inseto") {
            if (c == "hematofago") {
                printf("pulga\n");
            } else {
                printf("lagarta\n");
            }
        } else {
            if (c == "hematofago") {
                printf("sanguessuga\n");
            } else {
                printf("minhoca\n");
            }
        }
    }

    return 0;
}

File: solutions/beecrowd/1049/1049.py
import sys

classification_table = {
    'vertebrado': {
        'ave': {
            'carnivoro': 'aguia',
            'onivoro': 'pomba',
        },
        'mamifero': {
            'onivoro': 'homem',
            'herbivoro': 'vaca',
        },
    },
    'invertebrado': {
        'inseto': {
            'hematofago': 'pulga',
            'herbivoro': 'lagarta',
        },
        'anelideo': {
            'hematofago': 'sanguessuga',
            'onivoro': 'minhoca',
        },
    },
}

for body in sys.stdin:
    group = input()
    eating = input()

    print(classification_table[body.strip()][group][eating])
